### ISSUE TITLE
migrate away from bower

### ISSUE THREAD

**Issue by gruppjo** [mwaylabs/generator-m-ionic#488]
While I was investigating for our yarn integration https://github.com/mwaylabs/generator-m-ionic/issues/469 I stumbled across these two statements, that should motivate us to move away from bower in the long run.

- Migrating away from bower - https://github.com/yeoman/yeoman/issues/1521
- Bower is dead https://github.com/yeoman/generator/issues/964#issuecomment-258606541

@lordgreg, @MathiasTim. What do you think?

**Comment by lordgreg** [comment]
I'm against this issue since it would mean we would have to refactor so many things and take care of any other possible issues. This is a big change and should only be take care of if highly requested.

As for now, I wouldn't touch it at all.

**Comment by MathiasTim** [comment]
I'm thinking the same way @lordgreg does.
Also I'm not 100% sure, that the authors of the posts you mentioned are aware, that yarn does no more support bower. 
https://bower.io/blog/2016/using-bower-with-yarn/

I am fearing too big of a change from one generator release to another.

And btw. what does yeoman has to do with bower? We load yeoman over npm, don't we?

**Comment by gruppjo** [comment]
The part that could possibly impact the experience of the generator is that yeoman runs bower automatically for you after project generation. If they drop the support you'd have to run it separately. Just a minor inconvenience, though.

But you make valid arguments, it's probably a big change. However, I think it's good to stay on top of it and see how bower is supported in the long run.

Moving to npm yields some advantages two, on the other hand. You don't have to have bower installed as a separate dependency.

**Comment by MathiasTim** [comment]
> If they drop the support you'd have to run it separately. Just a minor inconvenience, though.
Yeah, this only happens for project setup, this happens to only a few devs sometimes a year so I'll be ok with that.

> But you make valid arguments, it's probably a big change, or is it? However, I think it's good to stay on top of it and see how bower is supported in the long run.
I think all the 'auto-injecting' and stuff will be different and can therefore be a lot work to do.  

> Moving to npm yields some advantages two, on the other hand. You don't have to have bower installed as a separate dependency.
True, but I think angular1 is still 'just inject the script into `index.html`'. And with npm you need to think about version conflicts, since bower is doing the resolutions and npm afaik doesn't, they just nest it. (flat vs nested dependency tree)
